### ISSUE TITLE
Batch Norm Layer weights/parameters turning NaNs during training of a Siamese Network with Triplet Loss

### ISSUE THREAD

**Issue by sidgairo18** [keras-team/keras#11897]
With reference to https://stackoverflow.com/questions/53400472/keras-model-weights-for-some-layers-become-all-nans , what I have noticed that the "NaNs" are present only in the "batch_norm" layers. Why must this be happening?

Is there some problem with the training - like Vanishing Gradient or Gradients exploding? or are the NaNs only placeholders for biases?

(Just a prior - I am trying to train a Siamese Network with Triplet Loss as mentioned here https://stackoverflow.com/questions/53400472/keras-model-weights-for-some-layers-become-all-nans (architecture and loss) ) 

Sample Weights listed here-
https://stackoverflow.com/questions/53830547/batch-norm-layer-weights-parameters-turning-nans-during-training-of-a-siamese-ne

Link to the Gist for the code I am using for training - https://gist.github.com/sidgairo18/dca347edd4588484237a231d7dab9a63

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [Yes] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [Yes] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [Yes] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by sidgairo18** [comment]
Is there a chance that there is a numerical instability matter associated with BN in Keras or Tensorflow as earlier experienced in Caffe.

Link : https://github.com/BVLC/caffe/pull/5136

**Comment by gabrieldemarmiesse** [comment]
Let's close this issue and track the bug with #11927.

**Comment by karolbadowski** [comment]
@gabrieldemarmiesse Hello. 
I see the issues closed one by one, but it seems that for 2 years still nonone has fixed the issue.
Is anyone working on the topic?

I also experience this problem in 2020 and first mentions of the problem are from 2018.

I believe there is a fundamental problem with how shared vision model instances not aggregate correctly...
If that is the case, maybe more things are going wrong abut data exchange between shared layers instances.
It is worth to investigate with slightly higher priority.

I fear that if tasks are closed one by one, noone will notice and take care of the persisting, reoccurring issue.
